src/Database/Type/DateTimeType.php

```diff
@@ -53,9 +53,12 @@ class DateTimeType extends Type implements TypeInterface
     /**
      * String format to use for DateTime parsing
      *
-     * @var string
+     * @var string|array
      */
-    protected $_format = 'Y-m-d H:i:s';
+    protected $_format = [
+        'Y-m-d H:i:s',
+        'Y-m-d\TH:i:s.uO',
```

Comment:
It should be fully BC as it only adds on top. But sure, we can also either make it opt-in from app level, or do it in a minor.
Maybe there are also even better options here than this.